java/bulk-import/bulk-import-starter/src/main/java/sleeper/bulkimport/starter/executor/BulkImportArguments.java

```diff
@@ -50,21 +51,31 @@ public List<String> constructArgs(String taskId, String jarLocation) {
         return constructArgs(bulkImportJob, taskId, jarLocation);
     }
 
-    public List<String> constructArgs(BulkImportJob bulkImportJob, String taskId, String jarLocation) {
+    public List<String> constructArgs(BulkImportJob bulkImportJob, String taskId, boolean isServerlessJob) {
         Map<String, String> userConfig = bulkImportJob.getSparkConf();
-        LOGGER.info("Using Spark config {}", userConfig);
 
         String className = bulkImportJob.getClassName() != null ? bulkImportJob.getClassName() : instanceProperties.get(BULK_IMPORT_CLASS_NAME);
 
-        List<String> args = Lists.newArrayList("spark-submit", "--deploy-mode", "cluster", "--class", className);
+        List<String> args = new ArrayList<>();
+
+        if (!isServerlessJob) {
+            args = Lists.newArrayList("spark-submit", "--deploy-mode", "cluster");
+        }
```

Comment:
This can be prefixed in the other method rather than adding a parameter to this one.

Reply:
Updated to the other method